### PR TITLE
[chrome-remote-interface] add initial definition

### PR DESCRIPTION
**Pull request by kazarmy**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [X] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [X] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [X] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [X] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [X] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [X] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

These are initial (incomplete) type definitions for [cyrus-and/chrome-remote-interface](https://github.com/cyrus-and/chrome-remote-interface). The Chrome Debugging Protocol (CDP) used should be the latest from [ChromeDevTools/devtools-protocol](https://github.com/ChromeDevTools/devtools-protocol), modified by the [`protocol-dts-generator.ts`](https://github.com/kazarmy/devtools-protocol/blob/master/scripts/protocol-dts-generator.ts) script at [kazarmy/devtools-protocol](https://github.com/kazarmy/devtools-protocol) so that it passes `tslint` and to add `Promise` event methods. The CDP license and version are included. Not sure (if this pr is accepted) whether to update the included CDP weekly, monthly or as needed.

### CODE REVIEW

File: types/chrome-remote-interface/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+    interface Client {
+        close: () => Promise<void>;
+        Page: ProtocolProxyApi.PageApi;
+        Runtime: ProtocolProxyApi.RuntimeApi;
+    }
+}
+
+declare const CDP: {
+    (options?: CDP.BaseOptions): Promise<CDP.Client>;
```
Comment (peterblazejewicz) [inline types/chrome-remote-interface/index.d.ts:32]: Should there be a `CDP` version with callback function?:
https://unpkg.com/browse/chrome-remote-interface@0.30.1/index.js

Comment (kazarmy) [reply types/chrome-remote-interface/index.d.ts:34]: Yes 53e2027 sorry oversight ... was exclusively looking at a different file ([lib/devtools.js](https://github.com/cyrus-and/chrome-remote-interface/blob/v0.30.1/lib/devtools.js)) than the one above.

The callback type definition is based on https://github.com/cyrus-and/chrome-remote-interface/blob/v0.30.1/index.js#L19. I don't see an attachment to the `error` event, just [a comment](https://github.com/cyrus-and/chrome-remote-interface/blob/v0.30.1/index.js#L15) stating that the error callback can be registered later. Not sure how connection errors (e.g. incorrect port) can be handled though if so.

**Comment by peterblazejewicz** [comment]
@Kolahzary do you reallly need to create this DT package by copying content of 3rd party types? AFAIK there are only types exported by `devtools-protocol`,so why not use that as an advantage? 
What about using TS v3.8 and using direct type imports and using `devtools-protocol` as proper dependency (which is in fact at runtime for JS usage)?
https://gist.github.com/peterblazejewicz/4a9dcefa11477156958a9c4ef8ecca34
in your `index.d.ts`:
```ts
// Minimum TypeScript Version: 3.8

import type ProtocolProxyApi from 'devtools-protocol/types/protocol-proxy-api';

declare namespace CDP {
```
let me know what do you think

**Comment by kazarmy** [comment]
@peterblazejewicz Apart from the need to convince the `devtools-protocol` team to accept a PR adding `Promise` event methods (which might be no small feat), from an engineering standpoint I think what you suggest is ideal and should be done.

> @Kolahzary ...

Emm I'm not the person above and I'm not based where that person is based. In fact, where I'm based has a pretty good human rights ranking just in case you're wondering. Of course there's the general wariness that what people say should happen and what actually happens can be different especially on a case-by-case basis, but that's true anywhere.

**Comment by peterblazejewicz** [comment]
sorry for autosuggestion :)

**Review by peterblazejewicz — APPROVED** [review]
LGTM!

**Review by peterblazejewicz — APPROVED** [review]

**Comment by westy92** [comment]
These are great! I'll submit a PR to improve these once merged. 👍 🎉 🚀

**Comment by kazarmy** [comment]
Ready to merge

**Comment by kazarmy** [comment]
> ... I'll submit a PR to improve these once merged.

Ok then, go for it! :smiley:

**Comment by westy92** [comment]
@kazarmy https://github.com/DefinitelyTyped/DefinitelyTyped/pull/54022 :)

**Comment by kazarmy** [comment]
Thanks! So far it looks good with regard to the source and README.md, but haven't checked it line-by-line yet, though will probably finish that by this weekend or sooner.

**Pull request merged**